MonoGame.Framework/Platform/Native/Interop.cs

```diff
@@ -0,0 +1,88 @@
+
+using Microsoft.Xna.Framework.Graphics;
+using System.Runtime.CompilerServices;
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+
+namespace MonoGame.Interop;
+
+internal readonly struct Game { }
+
+internal readonly struct GameWindow { }
+
+internal static unsafe partial class GameWrapper
+{
+    [LibraryImport("monogame", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void MG_GW_SetAllowUserResizing(Game* game, GameWindow* gameWindow, [MarshalAs(UnmanagedType.U1)] bool allowuserresizing);
+}
+
+internal struct PtrTo<T>
+{
+    public unsafe T* Ptr;
+}
+
+internal readonly struct MGG_GraphicsDevice { }
+internal readonly struct MGG_Buffer { }
+internal readonly struct MGG_Texture { }
+
+internal struct MGG_GraphicsDevice_Caps
+{
+    public int MaxTextureSlots;
+    public int MaxVertexTextureSlots;
+    public int MaxVertexBufferSlots;
+}
+
+internal static unsafe partial class MGG
```

Comment:
We need to wart these with something that works for us in C# and in C++ (don't want conflicts with other native libs).

So i'm thinking:

- MGG for Graphics.
- MGA for Audio.
- MGI for Input.
- MGF for Framework?

Thoughts?